src/GraphQLParser.Tests/Values/GraphQLFloatValueTests.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using GraphQLParser.AST;
+using Shouldly;
+using Xunit;
+
+namespace GraphQLParser.Tests;
+
+public class GraphQLFloatValueTests
+{
+    [Fact]
+    public void NoValue()
+    {
+        var value = new GraphQLFloatValue();
+        value.Value.Length.ShouldBe(0);
+    }
+
+    [Fact]
+    public void NanValue()
+    {
+        var ex = Should.Throw<ArgumentOutOfRangeException>(() => new GraphQLFloatValue(double.NaN));
+        ex.Message.ShouldStartWith("Value cannot be NaN.");
+    }
```

Comment:
Duplicate for Infinity and NegativeInfinity

Reply:
done 30 seconds ago :)